entrypoint.py

```
@@ -8,12 +8,22 @@
 from argparse import ArgumentParser
```

A few notes:

- I feel like this file is getting complex enough to be split into two. I think having one file for arg parsing and main, and another for the logic would be a good idea. This isn't necessary, so feel free to keep it as-is.
- I would prefer `search_strategy` to be an enum instead of a string. Apart from standard benefits like typing, knowing exactly what values are allowed etc, this could also lead to a "cleaner" refactor for `query_needs_manufacturer` etc. Like:

  ```py

  class SearchStrategy(Enum):
      FUZZY = "fuzzy"
      # ...

      def query_needs_manufacturer(self) -> bool:
          # ...
  ```

  The other advantage of this approach is that it better validates the input and fails fast if the user inputs an incorrect value.

---

I kind of thought that if Cofactr ever added a new search strategy, and it was just a string that we passed through, users could start using it without waiting for us to update the action, which we would need to do for an enum.

---

That's fair. I still think we should have some kind of warning/validation for the types, as if someone is typing them in they could make a typo. If we want to allow values we don't expect, we can warn them about it in the logs to make it clear.

---

I can change it to an enum. It's unlikely they'll add new strategies, and if they do, it will be infrequently.

---

> * I feel like this file is getting complex enough to be split into two. I think having one file for arg parsing and main, and another for the logic would be a good idea. This isn't necessary, so feel free to keep it as-is.

To me, it's not that it needs to be split in two as much as there should be tests. Right now, we're testing it by running it in an action in another repo. But there's no assertions about what the results should actually be.

Having tests would also inform the design in terms of how to split it up into multiple functions.

I could add tests in this PR. I thought it was out of scope and that the action as a whole is still in great flux where tests would be wasted effort as we change things around. But this PR might've actually gone faster if we had tests.